clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -118,6 +124,67 @@ class GCEUtil {
     }
   }
 
+  static Image getImageFromArtifact(Artifact artifact,
+                                    Compute compute,
+                                    Task task,
+                                    String phase,
+                                    SafeRetry safeRetry,
+                                    GoogleExecutorTraits executor) {
+    if (artifact.getType() != ArtifactUtils.GCE_IMAGE_TYPE) {
+      throw new GoogleOperationException("Artifact to deploy to GCE must be of type ${ArtifactUtils.GCE_IMAGE_TYPE}")
+    }
+
+    def reference = artifact.getReference()
+    task.updateStatus phase, "Looking up image $reference..."
+
+    def result = safeRetry.doRetry(
+      {
+        return compute.getRequestFactory()
+          .buildGetRequest(new GenericUrl(reference))
+          .setParser(new JsonObjectParser(JacksonFactory.getDefaultInstance()))
+          .execute()
+          .parseAs(Image)
```

Comment:
Generally the pattern with `safeRetry` is to wrap the heavy-duty platform work in the retry and then process the resource after the fact. I don't think you'd want to retry this whole closure if the `parseAs` fails - we can do this locally and report our own error rather than retrying and generating more network traffic.